account_invoice_rounding/account.py

```diff
@@ -158,125 +152,68 @@ def _compute_swedish_rounding(self, cr, uid, invoice, amounts,
-            },
-            multi='all'),
-        'amount_tax': fields.function(
-            _amount_all,
-            digits_compute=dp.get_precision('Account'),
-            string='Tax',
-            store={
-                'account.invoice': (lambda self, cr, uid, ids, c={}: ids,
-                                    ['invoice_line'], 20),
-                'account.invoice.tax': (_get_invoice_tax, None, 20),
-                'account.invoice.line': (_get_invoice_line,
-                                         ['price_unit',
-                                          'invoice_line_tax_id',
-                                          'quantity',
-                                          'discount',
-                                          'invoice_id'], 20),
-            },
-            multi='all'),
-        'amount_total': fields.function(
-            _amount_all,
-            digits_compute=dp.get_precision('Account'),
-            string='Total',
-            store={
-                'account.invoice': (lambda self, cr, uid, ids, c={}: ids,
-                                    ['invoice_line'], 20),
-                'account.invoice.tax': (_get_invoice_tax, None, 20),
-                'account.invoice.line': (_get_invoice_line,
-                                         ['price_unit',
-                                          'invoice_line_tax_id',
-                                          'quantity',
-                                          'discount',
-                                          'invoice_id'], 20),
-            },
-            multi='all'),
-    }
-
-
-class AccountInvoiceLine(orm.Model):
+                    self.amount_untaxed -= line.price_subtotal
+            self.amount_total = self.amount_tax + self.amount_untaxed
+            swedish_rounding = self._compute_swedish_rounding(
+                self,
+                [self.id])
+            if swedish_rounding:
+                self.amount_total = swedish_rounding['amount_total']
+                if 'amount_tax' in swedish_rounding:
+                    self.amount_tax = swedish_rounding['amount_tax']
+                elif 'amount_untaxed' in swedish_rounding:
+                    self.amount_untaxed = (
+                        swedish_rounding['amount_untaxed'])
+
+    @api.one
+    def _get_rounding_invoice_line_id(self):
+        line_ids = self.env['account.invoice.line'].search(
+            [('invoice_id', '=', self.id),
+             ('is_rounding', '=', True)])
+        self.global_round_line_id = line_ids and line_ids[0] or False
+
+    global_round_line_id = fields.Many2one(
+        'account.invoice.line',
+        string='Invoice Line for total rounding',
+        compute=_get_rounding_invoice_line_id,
+        readonly=True)
+    amount_untaxed = fields.Float(
+        digits_compute=dp.get_precision('Account'),
+        string='Subtotal',
+        track_visibility='always',
+        compute=_compute_amount,
+        multi='all')
```

Comment:
You can remove `multi` attribute as using same method is enough.

    multiple fields can be computed at the same time by the same method, just use the same method 
    on all fields and set all of them: [1]

[1] https://www.odoo.com/documentation/8.0/reference/orm.html#fields